File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/ProjectSyncIssues.java
package com.tyron.builder.model;

import org.jetbrains.annotations.NotNull;

import java.util.Collection;

/**
 * Model for a project's {@link SyncIssue}s.
 *
 * <p>This model should be fetched last (after other models), in order to have all the SyncIssue's
 * collected and delivered.
 */
public interface ProjectSyncIssues {

    /** Returns issues found during sync. */
    @NotNull
    Collection<SyncIssue> getSyncIssues();
}

File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/v2/models/ProjectSyncIssues.kt
package com.tyron.builder.model.v2.models

import com.tyron.builder.model.v2.AndroidModel
import com.tyron.builder.model.v2.ide.SyncIssue

/**
 * Model for a project's [SyncIssue]s.
 *
 * This model should be fetched last (after other models), in order to have all the SyncIssue's
 * collected and delivered.
 */
interface ProjectSyncIssues: AndroidModel {
    /** Returns issues found during sync.  */
    val syncIssues: Collection<SyncIssue>
}
